app/assets/javascripts/modules/namespaces/components/new-form.js

```diff
@@ -53,12 +60,92 @@ export default {
     },
   },
 
+  validations: {
```

Comment:
Instead of doing all this, we should instead call a single endpoint that returns whether the given name is valid. This way:

- We have a single source of truth.
- We have a simpler client-side code.

Reply:
> Instead of doing all this, we should instead call a single endpoint that returns whether the given name is valid.

Do you mean for all the validations? Something like `/namespaces/valid`? How would the response be?

Reply:
In my mind, you would have something like this:

- You have a `/status` endpoint for each resource (this also includes the `registry` resource from #1396).
- The response is a JSON that looks like this:

```json
{
  exists: false,
  valid: true,
  msg: ""
}
```

If the resource does not exist (e.g. you are checking that the form contains valid inputs), then `exists` is false and the `valid` and `msg` fields have to be checked. Otherwise, if the resource exists (e.g. checking that the current registry is reachable), then `valid` and `msg` can be safely ignored (e.g. and then the client-side has to show an "this already exists" message or something like that).

All this is just on the top of my head, so I may be missing something 😁

Reply:
I'm confused now, are you talking about the `exists` endpoint or the whole validation approach?

Reply:
To recap our conversation: we are gonna keep the same behavior (field by field) on the client side but move all the validation to the back-end through a single `/status` endpoint. Right?

I'm just not sure about the `status` name. It can have so many meanings. But that's easy to change later. 😉

Reply:
Sure, there are other possibilities for the name, like `/validate` or something like that.